src/hooks/usePatchStatusSelect.ts

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect } from "react";
+import { PatchBuildVariant } from "gql/generated/types";
+
+export const usePatchStatusSelect = (
```

Comment:
i really, really like that you extracted the state logic of a component into a hook. when a component has complex state management, it can get messy quickly, e.g. the configure patch page. using hooks to encapsulate that messy state logic is a great way to separate the different concerns of a component.